actions/lib/ansible_base.py

```diff
@@ -43,10 +43,13 @@ def _parse_extra_vars(self):
                 var_list = [n.strip() for n in var_list]
 
                 # Separate @file vars from kwarg vars
+                json_vars = list()
                 file_vars = list()
                 kwarg_vars = list()
```

Comment:
Splitting the vars between file_vars and kwarg_vars like this could have unforseen ordering problems. If you specifically set one var before or after a file, that ordering is lost here.